packages/server/lib/utils/orb.ts

```diff
@@ -0,0 +1,40 @@
+import { billing } from '@nangohq/billing';
+import db from '@nangohq/database';
+import { freePlan, updatePlanByTeam } from '@nangohq/shared';
+import { Err, Ok } from '@nangohq/utils';
+
+import type { BillingCustomer, BillingSubscription, DBTeam, DBUser, Result } from '@nangohq/types';
+
+/**
+ * Creates an Orb customer (if it doesn't exist) and links it to the account plan
+ */
+export async function linkOrbCustomer(account: DBTeam, user: DBUser): Promise<Result<BillingCustomer>> {
+    const resUpsert = await billing.upsertCustomer(account, user);
+    if (resUpsert.isErr()) {
+        return Err(resUpsert.error);
+    }
+
+    const resUpdate = await updatePlanByTeam(db.knex, { account_id: account.id, orb_customer_id: resUpsert.value.id });
```

Comment:
those 2 queries shows that maybe our data model is not ideal. Is the `orb_customer_id` required in the plan table. Maybe we should have a self-contained billing table that links account id and orb id. Would make sense since we are planning to have a dedicated `billing` service